content/en/apps/guides/hosting/4.x/app-developer.md

```diff
@@ -114,32 +114,110 @@ To read more about how `docker-compose` works, be sure to read the [helpful dock
 
 This script brings a lot of benefits with it:
 * You only have to download one bash script
-* It has been tested to work on Windows, macOS and Linux
 * All compose files and images will be downloaded automatically for you
 * All networks, storage volumes and containers will be created 
 * A valid TLS certificate will be installed, allowing you to easily test on with CHT Android natively on a mobile device
+* An unused port is automatically chosen for you when creating a new project.  No more manually looking at your existing `.env` files!
 
-Currently, the CHT Docker Helper for 4.0 is in beta, and comes with some caveats:
-* it is not yet available in `master`
-* it does not account for port conflicts, so you need to manually resolve them yourself by editing the `*.env` files
-* upgrades are not yet supported
-* concurrently running multiple instances is not yet supported
+### Installing
 
 To get started using it:
 1. Clone the [CHT Core](https://github.com/medic/cht-core/) repo
-2. Checkout the `7852-cht-4.x-docker-helper` branch
-3. `cd` into the `cht-core/scripts/docker-helper-4.x/` directory
-4. Run the script with `./cht-docker-compose.sh`
+2. When you want to check for updates, just run `git pull orign` in the `cht-core` directory.
+
+If you want a more stand-alone version, you can `curl` the bash script directly, but you can't use `git` to easily update it then:
+
+```shell
+curl https://raw.githubusercontent.com/medic/cht-core/master/scripts/docker-helper-4.x/cht-docker-compose.sh
+```
+
+### Usage
+
+Always run the script from directory where it lives.  If you launch it from further up the directory three, relative paths will fail:
+
+| Do | Don't |
+|---|---|
+| `./cht-docker-compose.sh`|`./docker-helper-4.x/cht-docker-compose.sh`|
+
+#### Launching
+
+Run the script with:
+
+```
+./cht-docker-compose.sh
+```
+
+The first time you run, you will be prompted to create a new project.  Here's what that looks like:
+
+```shell
+./cht-docker-compose.sh
+Would you like to initialize a new project [y/N]? y
+How do you want to name the project? 4 OH The First
+Downloading compose files ... done 
+
+Creating network "4_oh_the_first-cht-net" with the default driver
+Creating my_first_cht_project-dir_cht-upgrade-service_1 ... done
+Starting project "4_oh_the_first". First run takes a while. Will try for up to five minutes........
+
+ -------------------------------------------------------- 
+
+  Success! "4_oh_the_first" is set up:
+
+    https://127-0-0-1.my.local-ip.co:10444/ (CHT)
+    https://127-0-0-1.my.local-ip.co:10444/_utils/ (Fauxton)
+
+    Login: medic
+    Password: password
+
+ -------------------------------------------------------- 
+
+ Have a great day! 
```

Comment:
```suggestion
Start existing project
    ./cht-docker-compose.sh ENV-FILE.env

Stop and keep project:
    ./cht-docker-compose.sh ENV-FILE.env stop

Stop and destroy all project data:
    ./cht-docker-compose.sh ENV-FILE.env destroy

https://docs.communityhealthtoolkit.org/apps/guides/hosting/4.x/app-developer/


 Have a great day! 
```

Not sure we actually need to include all of this, but otherwise we should probably drop the "Have a great day!" part and just stick with the instance info here...